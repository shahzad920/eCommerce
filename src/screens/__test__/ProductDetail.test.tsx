/**
 * @format
 */

import 'react-native';
import React from 'react';
import {ProductDetail} from '..';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('Product Detail Renders Correctly', () => {
  const route = {
    params: {
      product: {
        img: '',
        name: 'Product',
        price: '891',
      },
    },
  };
  const tree = renderer.create(<ProductDetail route={route} />);

  expect(tree.toJSON()).toMatchSnapshot();
});
