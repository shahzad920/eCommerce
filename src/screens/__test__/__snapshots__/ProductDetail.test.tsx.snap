// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product Detail Renders Correctly 1`] = `
<View
  collapsable={false}
  style={
    {
      "backgroundColor": "rgba(247, 243, 249, 1)",
      "borderRadius": 12,
      "flex": 1,
      "shadowColor": "#000",
      "shadowOffset": {
        "height": 1,
        "width": 0,
      },
      "shadowOpacity": 0.15,
      "shadowRadius": 3,
    }
  }
  testID="Product-Detail-container-outer-layer"
>
  <View
    collapsable={false}
    style={
      {
        "backgroundColor": "rgba(247, 243, 249, 1)",
        "borderRadius": 12,
        "elevation": 1,
        "flex": undefined,
        "shadowColor": "#000",
        "shadowOffset": {
          "height": 1,
          "width": 0,
        },
        "shadowOpacity": 0.3,
        "shadowRadius": 1,
      }
    }
    testID="Product-Detail-container"
  >
    <View
      style={
        [
          {
            "flexShrink": 1,
          },
          undefined,
        ]
      }
      testID="Product-Detail"
    >
      <View
        style={
          [
            {
              "backgroundColor": "#eeeeee",
              "height": 195,
              "overflow": "hidden",
            },
            {
              "borderRadius": 12,
            },
            undefined,
          ]
        }
      >
        <Image
          accessibilityIgnoresInvertColors={true}
          borderRadiusStyles={{}}
          siblings={
            [
              "Card.Cover",
              "Card.Title",
              "Card.Actions",
            ]
          }
          source={
            {
              "uri": "",
            }
          }
          style={
            [
              {
                "flex": 1,
                "height": undefined,
                "justifyContent": "flex-end",
                "padding": 16,
                "width": undefined,
              },
              {
                "borderRadius": 12,
              },
            ]
          }
        />
      </View>
      <View
        style={
          [
            {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "paddingLeft": 16,
            },
            {
              "minHeight": 72,
            },
            undefined,
          ]
        }
      >
        <View
          style={
            [
              {
                "flex": 1,
                "flexDirection": "column",
                "justifyContent": "center",
              },
            ]
          }
        >
          <Text
            numberOfLines={2}
            style={
              [
                {
                  "textAlign": "left",
                },
                {
                  "color": "rgba(28, 27, 31, 1)",
                  "writingDirection": "ltr",
                },
                [
                  {
                    "fontFamily": "System",
                    "fontSize": 16,
                    "fontWeight": "400",
                    "letterSpacing": 0.15,
                    "lineHeight": 24,
                  },
                  [
                    {
                      "minHeight": 30,
                      "paddingRight": 16,
                    },
                    {
                      "marginBottom": 0,
                    },
                    undefined,
                  ],
                ],
              ]
            }
          >
            Product
          </Text>
          <Text
            numberOfLines={1}
            style={
              [
                {
                  "textAlign": "left",
                },
                {
                  "color": "rgba(28, 27, 31, 1)",
                  "writingDirection": "ltr",
                },
                [
                  {
                    "fontFamily": "System",
                    "fontSize": 14,
                    "fontWeight": "400",
                    "letterSpacing": 0.25,
                    "lineHeight": 20,
                  },
                  [
                    {
                      "marginVertical": 0,
                      "minHeight": 20,
                      "paddingRight": 16,
                    },
                    undefined,
                  ],
                ],
              ]
            }
          >
            Price: $891
          </Text>
        </View>
        <View />
      </View>
      <View
        borderRadiusStyles={{}}
        index={2}
        siblings={
          [
            "Card.Cover",
            "Card.Title",
            "Card.Actions",
          ]
        }
        style={
          [
            {
              "alignItems": "center",
              "flexDirection": "row",
              "padding": 8,
            },
            undefined,
            {
              "justifyContent": "flex-end",
            },
          ]
        }
        total={3}
      >
        <View
          collapsable={false}
          style={
            {
              "backgroundColor": "transparent",
              "borderRadius": 20,
              "marginLeft": 8,
              "shadowColor": "#000",
              "shadowOffset": {
                "height": 0,
                "width": 0,
              },
              "shadowOpacity": 0,
              "shadowRadius": 0,
            }
          }
          testID="Button-Add-To-Cart-container-outer-layer"
        >
          <View
            collapsable={false}
            style={
              {
                "backgroundColor": "transparent",
                "borderColor": "rgba(121, 116, 126, 1)",
                "borderRadius": 20,
                "borderStyle": "solid",
                "borderWidth": 1,
                "flex": undefined,
                "minWidth": 64,
                "shadowColor": "#000",
                "shadowOffset": {
                  "height": 0,
                  "width": 0,
                },
                "shadowOpacity": 0,
                "shadowRadius": 0,
              }
            }
            testID="Button-Add-To-Cart-container"
          >
            <View
              accessibilityRole="button"
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": false,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                [
                  {
                    "overflow": "hidden",
                  },
                  {
                    "borderRadius": 20,
                  },
                ]
              }
              testID="Button-Add-To-Cart"
            >
              <View
                style={
                  [
                    {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "justifyContent": "center",
                    },
                    undefined,
                  ]
                }
              >
                <Text
                  numberOfLines={1}
                  selectable={false}
                  style={
                    [
                      {
                        "textAlign": "left",
                      },
                      {
                        "color": "rgba(28, 27, 31, 1)",
                        "writingDirection": "ltr",
                      },
                      [
                        {
                          "fontFamily": "System",
                          "fontSize": 14,
                          "fontWeight": "500",
                          "letterSpacing": 0.1,
                          "lineHeight": 20,
                        },
                        [
                          {
                            "marginHorizontal": 16,
                            "marginVertical": 9,
                            "textAlign": "center",
                          },
                          false,
                          {
                            "marginHorizontal": 24,
                            "marginVertical": 10,
                          },
                          false,
                          false,
                          {
                            "color": "rgba(103, 80, 164, 1)",
                            "fontFamily": "System",
                            "fontSize": 14,
                            "fontWeight": "500",
                            "letterSpacing": 0.1,
                            "lineHeight": 20,
                          },
                          undefined,
                        ],
                      ],
                    ]
                  }
                  testID="Button-Add-To-Cart-text"
                >
                  Add To Cart
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
